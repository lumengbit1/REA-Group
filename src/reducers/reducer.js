import { handleActions } from 'redux-actions';
import { produce } from 'immer';
import { normalize, schema } from 'normalizr';
import { remove } from 'lodash';
import {
  get_successed,
  get_failed,
  get,
  get_loading,
  add_property,
  remove_property,
} from './actions';

const initialState = {
  result: {
    results: [],
    saved: [],
  },
  entities: {},
  errors: {},
  loading: undefined,
};

const getReducer = handleActions(
  {
    [get]: (state) => state,
    [get_loading]: (state, action) => produce(state, (draft) => {
      draft.loading = action.payload.data;
    }),
    [get_successed]: (state, action) => produce(state, (draft) => {
      const results = new schema.Entity('results');
      const saved = new schema.Entity('saved');
      const value = {
        results: [results],
        saved: [saved],
      };
      const normalizedData = normalize(action.payload.data, value);

      draft.entities = { ...draft.entities, ...normalizedData.entities.results, ...normalizedData.entities.saved };
      draft.result.results = normalizedData.result.results;
      draft.result.saved = normalizedData.result.saved;
    }),
    [get_failed]: (state, action) => produce(state, (draft) => { draft.errors = action.payload.data; }),
    [add_property]: (state, action) => produce(state, (draft) => {
      remove(draft.result.results, (n) => n === action.payload);

      draft.result.saved.push(action.payload);
    }),
    [remove_property]: (state, action) => produce(state, (draft) => {
      remove(draft.result.saved, (n) => n === action.payload);

      draft.result.results.push(action.payload);
    }),
  },
  initialState,
);

export default getReducer;
