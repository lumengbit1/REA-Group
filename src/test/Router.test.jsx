import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import { createStore, applyMiddleware } from 'redux';
import { combineReducers } from 'redux-immutable';
import Router from '../router/routers';
import reducer from '../reducers';

const store = createStore(reducer, applyMiddleware(thunk));

describe('Route Test', () => {
  it('Route Test', () => {
    const component = renderer
      .create(
        <Provider store={store}>
          <MemoryRouter>
            <Router />
          </MemoryRouter>
        </Provider>,
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
