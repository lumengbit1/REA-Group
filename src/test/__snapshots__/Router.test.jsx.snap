// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Route Test Route Test 1`] = `
<div
  className="sc-iBPTik dPsWSS"
>
  <div
    className="sc-pGacB OBVcn"
  >
    <div
      className="sc-fubCzh dsyJIs"
    >
      Results
    </div>
    <div
      className="sc-jrAFXE cjmwOK"
    />
  </div>
  <div
    className="sc-pGacB OBVcn"
  >
    <div
      className="sc-fubCzh dsyJIs"
    >
      Saved Properties
    </div>
    <div
      className="sc-jrAFXE cjmwOK"
    />
  </div>
</div>
`;
